jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/servlets/ActionSPARQL.java

```diff
@@ -205,7 +206,11 @@ public static void parse(HttpAction action, StreamRDF dest, InputStream input, L
                 .lang(lang)
                 .base(base)
                 .parse(dest);
-        } 
+        } catch (RuntimeIOException ex) {
+            if ( ex.getCause() instanceof CharacterCodingException )
+                throw new RiotException("Character Coding Error: "+ex.getMessage());
```

Comment:
maybe `throw new RiotException("Character Coding Error: "+ex.getMessage(), ex.getCause());`  to keep more context?